AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/SyncMutationToCloudOperation.swift

```diff
@@ -51,14 +66,12 @@ class SyncMutationToCloudOperation: Operation {
         }
 
         log.debug(#function)
-        guard let api = api else {
-            // TODO: This should be part of our error handling routines
-            log.error("\(#function): API unexpectedly nil")
-            let apiError = APIError.unknown("API unexpectedly nil", "")
-            finish(result: .failed(apiError))
-            return
+        if let apiRequest = createAPIRequest() {
+            makeAPIRequest(apiRequest)
         }
```

Comment:
If the request is `nil` do we need to recover or propagate an error?

Reply:
oup, good call out, for now, lets fail, and if we need to build out a recovery mechanism, we can at a later time.